### PR TITLE
init api, db, di

### PR DESCRIPTION
**Pull request by hungld-1302**
## Related Tickets
- [#Task 56365: Initial API,database, DI](https://edu-redmine.sun-asterisk.vn/issues/xxxx)
## WHAT
- Create database, setup RoomDB
- Setup Retrofit, handle call API
- Setup DI
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

**Review by nhanviet1 — APPROVED** [review]

**Review by ndt0409 — APPROVED** [review]

File: gradle.properties [hunk 1]
```diff
@@ -7,6 +7,7 @@
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
 org.gradle.jvmargs=-Xmx2048m -Dfile.encoding=UTF-8
+BASE_URL = https://www.thecocktaildb.com/api/json/v1/1/
```
Comment (minhnn-1279) [inline gradle.properties]: skip working tree file này đi a :v

File: app/src/main/java/com/sildev/thecocktail/base/BaseAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.sildev.thecocktail.base
+
+import androidx.recyclerview.widget.RecyclerView
+
+abstract class BaseAdapter<T, VH : BaseViewHolder<T>> : RecyclerView.Adapter<VH>() {
```
Comment (vietnh-1109) [inline app/src/main/java/com/sildev/thecocktail/base/BaseAdapter.kt]: project này dùng list adapter nhé

File: app/src/main/java/com/sildev/thecocktail/base/BaseViewHolder.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    init {
+        itemView.setOnClickListener { itemData?.let { onItemClickListener(it) } }
+    }
+
+    open fun onBindData(itemData: T) {
+        this.itemData = itemData
+    }
+
+    open fun onItemClickListener(itemData: T) = onClickItem(itemData)
```
Comment (vietnh-1109) [inline app/src/main/java/com/sildev/thecocktail/base/BaseViewHolder.kt]: hàm này để làm gì , sao ko gọi luôn onClickItem()

File: app/src/main/java/com/sildev/thecocktail/data/model/Category.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.sildev.thecocktail.data.model
+
+import com.google.gson.annotations.SerializedName
+
+data class Category(
+    @SerializedName("strCategory")
+    val name: String
```
Comment (vietnh-1109) [inline app/src/main/java/com/sildev/thecocktail/data/model/Category.kt]:
```suggestion
    val name: String?
```
nhớ check null khi. sử dụng gson converter

**Pull request merged**